addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -630,7 +630,14 @@ export const editorCommands = {
         // Color the selected <font>s and remove uncolored fonts.
         const fontsSet = new Set(fonts);
         for (const font of fontsSet) {
-            colorElement(font, color, mode);
+            const isHexFormat = color.startsWith('#');
+            if(isHexFormat && opacity) {
+                const hexOpacity = '99';
```

Comment:
Thanks for the review! That is obvious idea one can think of in the first place. But in our case we have some cases in which we don't want to apply the opacity. e.g. if we set the background color of selected cells in a table OR trying to apply `foreColor` then opacity should not be applied. This can be handled easily in this method as it is more editor stuff than `color_palette.js`.  Also, the eventName `foreColor` or `backColor` is decided in `wysiwyg.js` file. Morover, right now `color_palette.js` gets the information of color from `_getButtonInfo` method which gets the information from dataset of `buttonEl` which is probably a more static way. Hence IMO we should add opacity at some point.